enterprise/coordinator/history/configmapstore.go

```diff
@@ -0,0 +1,234 @@
+// Copyright 2025 Edgeless Systems GmbH
+// SPDX-License-Identifier: AGPL-3.0-only
+
+//go:build enterprise
+
+package history
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log/slog"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+)
+
+// timeout is the timeout for Kubernetes API calls.
+const timeout = 10 * time.Second
+
+var keyRe = regexp.MustCompile(`^[a-zA-Z0-9-]+/[a-zA-Z0-9-]+$`)
+
+// ConfigMapStore is a Store implementation backed by Kubernetes Config Maps.
+type ConfigMapStore struct {
+	client    kubernetes.Interface
+	namespace string
+	uid       types.UID
+	logger    *slog.Logger
+}
+
+// NewConfigMapStore creates a new instance backed by Kubernetes Config Maps.
+func NewConfigMapStore(client kubernetes.Interface, log *slog.Logger) (*ConfigMapStore, error) {
+	namespace, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
```

Comment:
Let's make the namespace an input, that way you should be able to test with `NewConfigMapStore` instead of creating the struct directly.